presto-main/src/main/java/com/facebook/presto/sql/gen/CursorProcessorCompiler.java

```diff
@@ -75,13 +94,30 @@ public void generateMethods(SqlFunctionProperties sqlFunctionProperties, ClassDe
 
         generateProcessMethod(classDefinition, projections.size());
 
-        Map<LambdaDefinitionExpression, CompiledLambda> filterCompiledLambdaMap = generateMethodsForLambda(classDefinition, callSiteBinder, cachedInstanceBinder, filter, metadata, sqlFunctionProperties, "filter");
-        generateFilterMethod(sqlFunctionProperties, classDefinition, callSiteBinder, cachedInstanceBinder, filterCompiledLambdaMap, filter);
+        ImmutableMap.Builder<LambdaDefinitionExpression, CompiledLambda> compiledLambdaMapBuilder = ImmutableMap.builder();
+        compiledLambdaMapBuilder.putAll(generateMethodsForLambda(classDefinition, callSiteBinder, cachedInstanceBinder, filter, metadata, sqlFunctionProperties, "filter"));
+
+        for (int i = 0; i < projections.size(); i++) {
+            String methodName = "project_" + i;
+            compiledLambdaMapBuilder.putAll(generateMethodsForLambda(classDefinition, callSiteBinder, cachedInstanceBinder, projections.get(i), metadata, sqlFunctionProperties, methodName));
```

Comment:
Why is this needed (handle lambda differently from before)? Ideally we'd want to reuse lambda bytecode if they are the same cross different projections. If there's any issue preventing us from doing so, we should try to fix that first.

Reply:
@rongrong This is actually the original implementation. It is changed to generate Lambda methods from projections and filter at once with `methodName=""`

Reply:
This is not necessary. `LambdaBytecodeGenerator` has an API to take `List<RowExpression>`. In fact the API that takes a single `RowExpression` internally calls the other API. So we might as well change this logic to directly generate lambdas for all projections in one go.

Reply:
@rongrong Exactly, that's how it is implemented now. The comment is based on an outdated commit